tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,215 @@
+            };
+
+            engPathOption = new Option<string>(
+                "--eng-path")
+            {
+                Description = "Path to an 'eng' folder within an SDK repo (ex: /home/user/src/azure-sdk-for-go/eng)",
+                IsRequired = true,
+            };
+
+            outputPathOption = new Option<string>(
+                "--output-path")
+            {
+                Description = "Path to write the generated README contents",
+                IsRequired = true,
+            };
+
+            outputPathOption.SetDefaultValue("README.output.md");
+
+            modelOption = new Option<string>("--model")
+            {
+                Description = "The OpenAI model to use when generating the readme. Note, this will match the name of your Azure OpenAI model deployment.",
+                IsRequired = true,
+            };
+
+            modelOption.SetDefaultValue("gpt-4.1");
+
+            var command = new Command("readme", "README generator tool") {
+                engPathOption,
+                modelOption,
+                outputPathOption,
+                packagePathOption,
+                serviceDocumentationOption,
+                templatePathOption,
+            };
+
+            command.SetHandler(async (ic) => await HandleCommand(ic, ic.GetCancellationToken()));
+
+            return command;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var pr = ctx.ParseResult;
+
+            var templatePath = pr.GetValueForOption(templatePathOption);
+            var serviceDocumentation = pr.GetValueForOption(serviceDocumentationOption);
+            var packagePath = pr.GetValueForOption(packagePathOption);
+            var outputPath = pr.GetValueForOption(outputPathOption);
+            var engPath = pr.GetValueForOption(engPathOption);
+            var model = pr.GetValueForOption(modelOption);
+
+            var chatClient = openAiClient.GetChatClient(model);
+
+            await Generate(
+                chatClient: chatClient,
+                engPath: engPath,
+                templatePath: templatePath,
+                serviceDocumentation: new Uri(serviceDocumentation),
+                packagePath: packagePath,
+                outputPath: outputPath);
+        }
+
+        async Task Generate(ChatClient chatClient, string templatePath, Uri serviceDocumentation, string packagePath, string outputPath, string engPath)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(templatePath);
+            ArgumentNullException.ThrowIfNull(serviceDocumentation);
+            ArgumentException.ThrowIfNullOrEmpty(outputPath);
+            ArgumentException.ThrowIfNullOrEmpty(engPath);
+
+            var readmeText = await File.ReadAllTextAsync(templatePath);
+
+            var messages = new ChatMessage[]{
+                new SystemChatMessage(
+                    string.Join("\n",
```

Comment:
This is not a pattern that you want.   You're allocating like crazy here.   You can avoid this by just using a literal string.

```csharp
new SystemChatMessage("""
  Hello, my name is Richard and I'm still sad that we 
  never built the receiver redirect feature because it 
  is my friend Chris' favorite thing in the world
  and we let him down.
""");
```

Reply:
I added that to the prompt.

Reply:
It's beautiful now.